From 31bebbef0cce2ae68c25a2b0cbfc040f938791e9 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 4 May 2020 15:13:24 +0100
Subject: [PATCH] overlays: Make use of intra-overlay fragments

The firmware and runtime overlay support has recently been updated to
correctly process fragments that target other fragments within the
overlay. Make use of that ability and avoid the use of the awkward
target-path = "<alias>/..." workaround and for better readability.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/ads1015-overlay.dts | 8 ++++----
 arch/arm/boot/dts/overlays/ads1115-overlay.dts | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

--- a/arch/arm/boot/dts/overlays/ads1015-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ads1015-overlay.dts
@@ -24,7 +24,7 @@
     };
 
     fragment@1 {
-        target-path = "i2c_arm/ads1015";
+        target = <&ads1015>;
         __overlay__ {
             #address-cells = <1>;
             #size-cells = <0>;
@@ -37,7 +37,7 @@
     };
 
     fragment@2 {
-        target-path = "i2c_arm/ads1015";
+        target = <&ads1015>;
         __dormant__ {
             #address-cells = <1>;
             #size-cells = <0>;
@@ -50,7 +50,7 @@
     };
 
     fragment@3 {
-        target-path = "i2c_arm/ads1015";
+        target = <&ads1015>;
         __dormant__ {
             #address-cells = <1>;
             #size-cells = <0>;
@@ -63,7 +63,7 @@
     };
 
     fragment@4 {
-        target-path = "i2c_arm/ads1015";
+        target = <&ads1015>;
         __dormant__ {
             #address-cells = <1>;
             #size-cells = <0>;
--- a/arch/arm/boot/dts/overlays/ads1115-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ads1115-overlay.dts
@@ -26,7 +26,7 @@
 	};
 
 	fragment@1 {
-		target-path = "i2c_arm/ads1115";
+		target = <&ads1115>;
 		__dormant__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -40,7 +40,7 @@
 	};
 
 	fragment@2 {
-		target-path = "i2c_arm/ads1115";
+		target = <&ads1115>;
 		__dormant__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -54,7 +54,7 @@
 	};
 
 	fragment@3 {
-		target-path = "i2c_arm/ads1115";
+		target = <&ads1115>;
 		__dormant__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -68,7 +68,7 @@
 	};
 
 	fragment@4 {
-		target-path = "i2c_arm/ads1115";
+		target = <&ads1115>;
 		__dormant__ {
 			#address-cells = <1>;
 			#size-cells = <0>;
